### DISCUSSION TITLE
Is it possible to call fetch and load data within a docusaurs page / document

### DISCUSSION THREAD

**Discussion by WillFrasier** [facebook/docusaurus#4499] [General]
I want to render the output of a web request using the fetch() api. This request needs to be secured (so AAD auth). 

Is that possible in docusaurus?

**Comment by slorber** [comment]
It is possible to call an authenticated api client-side using `useEffect` or any lib like react-query/swr/react-async-hook and display some sensitive data.

You'll need to make sure that those data are not server-rendered, and use the BrowserOnly API for example:

https://docusaurus.io/docs/docusaurus-core#browseronly

I don't know what AAD auth is but anything you can do with Create-React-App should be doable with Docusaurus as well.

**Comment by armano2** [comment]
if by AAD auth you mean [Azure Active Directory authentication](https://docs.microsoft.com/en-us/azure/active-directory/authentication/overview-authentication) you can try using: https://www.npmjs.com/package/react-aad-msal
